### PR TITLE
fix(functions) + feat(new parsing)

### PR DESCRIPTION
**Pull request by devlordduck**
Just fixed 2 emoji funcs

### CODE REVIEW

**Comment by LordexDuck3990** [comment]
ISC9YzzlR4

File: src/handler/parsers.js [hunk 1]
```diff
@@ -533,42 +49,785 @@ const SlashOptionsParser = async (options) => {
...
+*/
+let checkBoolean = (text, orelse) => {
+    if (orelse && typeof orelse !== "boolean") return;
+    if (!text  || typeof text   !== "string" ) return orelse;
+
+    if ( text && text.trim() && (text.trim() === "" || text.trim() === " ") ) 
+        return orelse;
+
+    let yes = [ "true"  ];
+    let no  = [ "false" ];
```
Comment (Leref) [inline src/handler/parsers.js]: This completely can be changed into one array instead of two.

File: src/handler/parsers.js [hunk 2]
```diff
@@ -533,42 +49,785 @@ const SlashOptionsParser = async (options) => {
...
+
+        if (data && data.newMessageData && data.newMessageData.embeds) data.newMessageData.embeds.push(...embeds);
+        return embeds;
+      },
+    },
+    ComponentParser: {
+        name: ["actionRow", "Components", "newRow"],
+        code: async (d, data) => {
+            let code = mustEscape(data.part);
+            let nameofparserorsmtidk = "actionRow";
```
Comment (Leref) [inline src/handler/parsers.js]: Rename this variable

File: src/handler/parsers.js [hunk 3]
```diff
@@ -533,42 +49,785 @@ const SlashOptionsParser = async (options) => {
...
+        
+            for (var element of elements) {
+                element = element.slice(0, element.lastIndexOf("}"));
+        
+                const Checker = (name) => element.includes(`{${name}:`);
+
+                let components = [];
+        
+                if (Checker("button")) {
+                    var buttons = element.split("{button:").slice(1);
```
Comment (Leref) [inline src/handler/parsers.js]: Everything down here should be using let or const instead.

File: src/handler/parsers.js [hunk 4]
```diff
@@ -533,42 +49,785 @@ const SlashOptionsParser = async (options) => {
...
+
+      if (returnMsg === "id") {
+        return newMessage.id;
+      } else if (returnMsg === "object") {
+        return newMessage;
+      } else if (returnMsg === "withMessage") return newMessage;
     }
-    return optionData;
-};
-module.exports = {
```
Comment (Leref) [inline src/handler/parsers.js]: There is some files still using these methods

**Review by Leref — CHANGES_REQUESTED** [review]

File: src/handler/parsers.js [hunk 5]
```diff
@@ -533,42 +49,785 @@ const SlashOptionsParser = async (options) => {
...
+    if (returnMsg)
+      return {
+        content:
+          errorMessage.addBrackets() === "" ? " " : errorMessage.addBrackets(),
+        embeds: send ? newMessageData.embeds : [],
+        components: newMessageData.components,
+        files: newMessageData.files,
+        options: newMessageData.options,
+        reply: newMessageData.reply ? newMessageData.reply : undefined,
+        ...DataToReturnLol,
```
Comment (Leref) [inline src/handler/parsers.js]: Rename

**Comment by devlordduck** [comment]
done ig

**Comment by devlordduck** [comment]
yeah done

**Comment by Leref** [comment]
conflicts

**Pull request closed without merging**